### DISCUSSION TITLE
How important is exact contest scoring in MRCE?

### DISCUSSION THREAD

**Discussion by w7sst** [w7sst/MorseRunner#137] [General]
As I was wrapping up work on ARRL DX Contest, I noticed that we have ignored contest scoring as new contests have been added.

Contests seem to have a contest-specific scoring formula. To summarize our current contests (and I'm not trying to be precise):
Contest | Scoring Formula
---------|-------------------
ARRL DX | total **QSO points** * sum of **Zone and Country points**; where **QSO Points** is 3 points for inter-continent contacts, 1 point for same continent w/ different country, 2 points between different countries in NA, 0 points for contacts between stations in same country; **Zone and Country points** is sum of unique CQ-Zones worked plus sum of unique countries worked.
ARRL Field Day | 2 points per QSO (no multiplier)
CQ WPX, HST | number of QSOs * number of unique Prefixes
CQ WW | total QSO Points * sum of zone and country multipliers
CWOPS CWT | 1 point per QSO * number of callsigns worked (seems strange since only working one band; this results in N^2 resulting value)
NCJ NAQP | number of contacts * sum of number of states/provinces worked

My thoughts...
- Our focus is contest simulation and to me part of that can including the scoring.
- I don't want to code this scoring to be exact, I think approximate is okay for this.
- MRCE is not a results tabulation engine.
- HST mode must provide the same scoring result as 1.68 (which to my knowledge it does).
- In the future, if we allow posting of contest-specific scoring, it would be nice if the scoring matches current contest scoring rules.

I have the above working in code right now, but wanted to get a sense from each of you on how important this is. I think we can submit this knowing that the scoring result will not be guaranteed to be 100% accurate. Having a score following the spirit of the contest is probably sufficient.

Thoughts???

**Comment by scotthibbs** [comment]
I'm glad you made an issue for this. Figure if we can, we should. It helps keep the contesting more realistic with realistic scores. For example, in ARRL DX contest finding and answering calls over others means more points. Without accurate scoring, MR wouldn't reproduce or groom for this real world behavior.

**Comment by K6OK** [comment]
@w7sst -- I concur with your "approximate" approach.  It has to be.  WPX gives 1 point for contacts on high bands and 2 for low bands.  Obviously there's no frequency component to MRCE so that can't be done.  Field Day has a whole raft of bonus points depending on what kind of station is set up.  So I'd say exact contest scoring is not feasible, but simple approximations of scoring per contest is very worthwhile.

**Comment by scotthibbs** [comment]
Deleted previous comment that advertised nudes and reported behavior to github.
